Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -128,7 +131,12 @@ public Task<Page> PopAsync()
 
 			Task<Page> result = ((INavigationPageController)this).PopAsyncInner(animated);
 			CurrentNavigationTask = result;
-			return await result;
+			return await result.ContinueWith((t) =>
+			{
+				if (t.Status == TaskStatus.RanToCompletion)
+					UpdateActiveNavigationTaskCount();
+				return t.Result;
```

Comment:
In this case, we could also probably call `ResolveActiveNavigation` and then get the result.